testing/xfpga/test_events_c.cpp

```diff
@@ -182,19 +220,63 @@ int set_irq(mock_object * m, int request, va_list argp){
   goto out;
 }
 
+int set_uport_irq(mock_object * m, int request, va_list argp){
+  int retval = -1;
+  errno = EINVAL;
+  UNUSED_PARAM(m);
+  UNUSED_PARAM(request);
+  struct fpga_port_uafu_irq_set *uafu_irq =
+  	va_arg(argp, struct fpga_port_uafu_irq_set *);
+  if (!uafu_irq) {
+  	FPGA_MSG("uafu_irq is NULL");
+  	goto out_EINVAL;
+  }
+  if (uafu_irq->argsz < sizeof(*uafu_irq)) {
+  	FPGA_MSG("wrong structure size");
+  	goto out_EINVAL;
+  }
+  if (uafu_irq->flags != 0) {
+  	FPGA_MSG("unexpected flags %u", uafu_irq->flags);
+  	goto out_EINVAL;
+  }
+  if (gEnableIRQ) {
+  	uint32_t i;
+  	uint64_t data = 1;
+  	// Write to each eventfd to signal one IRQ event.
+  	for (i = 0 ; i < uafu_irq->count ; ++i) {
+  		if (uafu_irq->evtfd[i] >= 0)
+  			if (write(uafu_irq->evtfd[i], &data, sizeof(data)) !=
+  					sizeof(data)) {
+  				FPGA_ERR("IRQ write < 8 bytes");
+  			}
+  	}
+  }
+  retval = 0;
+  errno = 0;
+out:
+  va_end(argp);
+  return retval;
+
+out_EINVAL:
+  retval = -1;
+  errno = EINVAL;
+  goto out;
+}
+
 class events_p : public ::testing::TestWithParam<std::string> {
  protected:
   events_p()
       : tokens_dev_{{nullptr, nullptr}},
         tokens_accel_{{nullptr, nullptr}},
-        tmpfpgad_log_("tmpfpgad-XXXXXX.log"),
-        tmpfpgad_pid_("tmpfpgad-XXXXXX.pid"),
         handle_dev_(nullptr),
         handle_accel_(nullptr) {}
 
   virtual void SetUp() override {
-    tmpfpgad_log_ = mkstemp(const_cast<char *>(tmpfpgad_log_.c_str()));
-    tmpfpgad_pid_ = mkstemp(const_cast<char *>(tmpfpgad_pid_.c_str()));
+    strcpy(tmpfpgad_log_, "tmpfpgad-XXXXXX.log");
+    tmpfpgad_log_fd_ = mkstemps(tmpfpgad_log_, 4);
+    strcpy(tmpfpgad_pid_, "tmpfpgad-XXXXXX.pid");
+    tmpfpgad_pid_fd_ = mkstemps(tmpfpgad_pid_, 4);
```

Comment:
Let's not track the fds.  I convinced myself it's not necessary.  Instead..
  close(mkstemps(tmpfpgad_log_, 4));
  close(mkstemps(tmpfpgad_pid_, 4));